Bottom: 5cff9d29d388ff748c8e5abe052fe0003fca94d2
Top:    c6669ecaf6fa49e52318b8876bb6a6281e6ad469
Author: Parth Shah <parth@linux.ibm.com>
Date:   2019-10-07 14:00:46 +0530

sched/core: Add manual background task classification using sched_setattr syscall

Small background tasks typically performs some housekeeping work and are
less important in the overall scheme of load balancing and scheduling.

So provide a way to mark the task which are small background noises with
the use of additional flag to the existing task attribute. Also provide an
interface from the userspace which uses sched_setattr syscall to mark such
tasks.

The scheduler may use this as hints to pack such tasks on fewer number of
cores.

Signed-off-by: Parth Shah <parth@linux.ibm.com>

---

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index edae9277e48d..1b8280a7adcb 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -2497,3 +2497,7 @@ static inline void membarrier_switch_mm(struct rq *rq,
 {
 }
 #endif
+
+#define lenient_latency(lat)	((lat) == MAX_LATENCY_NICE)
+#define task_latency_lenient(p)	\
+	(lenient_latency((p)->latency_nice))
